src/main/java/duke/Tasks/TaskList.java

```diff
@@ -0,0 +1,57 @@
+package duke.Tasks;
+
+import duke.Exceptions.CommandException;
+import duke.Exceptions.DescriptionException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class TaskList extends ArrayList<Task> {
+    public TaskList() {
+        super();
+    }
+
+    public TaskList(BufferedReader br) throws CommandException, IOException {
+        super();
+        String str;
+        while ((str = br.readLine()) != null) {
+            this.add(Task.strToTask(str));
+        }
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder result = new StringBuilder();
+        for (Task task : this) {
+            result.append(task).append("\n");
+        }
+        return result.toString();
+    }
+
+    public String toFormattedString() {
+        int count = 1;
+        StringBuilder result = new StringBuilder();
+        for (Task task : this) {
+            result.append(count++)
```

Comment:
Maybe there is a better way you can do this? For example, you can try using String.format, though that would mean you need to change a bit more in your code. Small issue though, good work in general!